shell/platform/android/io/flutter/embedding/engine/android/FlutterView.java

```diff
@@ -300,6 +301,27 @@ public boolean onHoverEvent(MotionEvent event) {
   }
   //-------- End: Process UI I/O that Flutter cares about. ---------
 
+  //-------- Start: Accessibility -------
+  public void updateAccessibilityFeatures() {
+    // TODO(mattcarroll): get the following accessibility code working. This method is added now to
+    //                    avoid forgetting its invocation in FlutterFragment, but I want to think
+    //                    more deeply about how much accessibility logic should truly live in
+    //                    FlutterView vs AccessibilityBridge.
+//    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+//      String transitionAnimationScale = Settings.Global.getString(getContext().getContentResolver(),
+//          Settings.Global.TRANSITION_ANIMATION_SCALE);
+//      if (transitionAnimationScale != null && transitionAnimationScale.equals("0")) {
+//        accessibilityFeatureFlags ^= AccessibilityFeature.DISABLE_ANIMATIONS.value;
+//      } else {
+//        accessibilityFeatureFlags &= ~AccessibilityFeature.DISABLE_ANIMATIONS.value;
+//      }
+//    } else {
+//      // TODO(mattcarroll): we need to do something here for API 16
+//    }
+//    flutterEngine.getRenderer().setAccessibilityFeatures(accessibilityFeatureFlags);
```

Comment:
Why not just move this whole method to FlutterEngine?

Reply:
As far as I can tell, this method has nothing to do with a FlutterEngine.  If you're exposing iOS UIView and UIViewController accessibility behaviors in the generic concept of an engine, you might consider reversing course on that.  A FlutterEngine might run without any UI at all, so it doesn't seem logical to push view hierarchy responsibilities within that boundary.

Reply:
The engine can run without any UI for sure, but it has to know whether a View is available.  If the View is available and this event happens, it pushes that event to the view.  It may be more natural for the engine to contain this rather than the fragment needing to.  The View can't contain it entirely because all of this comes from Dart land, and the engine is what bridges Dart and the view.

Reply:
It's possible that we are using the term engine in very different ways.  Maybe that is why we are seeing these issues so differently.  As I mentioned above, the most productive thing to do is probably to chat about these problems in their own right and keep that independent of this particular PR.

Reply:
After discussing offline, this is fine for now - it's a bigger project to get this disentanged and this doesn't make us any worse than we already are.